imap_processing/swapi/l1/swapi_l1.py

```diff
@@ -1 +1,520 @@
+    plan_id = full_sweep_sci["PLAN_ID_SCIENCE"].data.reshape(-1, 12)
+    sweep_table = full_sweep_sci["SWEEP_TABLE"].data.reshape(-1, 12)
+
+    mode = full_sweep_sci["MODE"].data.reshape(-1, 12)
+
+    total_sweeps = len(plan_id)
+    bad_data_start_indices = []
+
+    for index in range(total_sweeps):
+        if not np.all(sweep_table[index] == sweep_table[index][0]):
+            logging.debug("SWEEP_TABLE is not same")
+            bad_data_start_indices.append(index)
+
+        if not np.all(plan_id[index] == plan_id[index][0]):
+            logging.debug("PLAN_ID is not same")
+            bad_data_start_indices.append(index)
+
+        # TODO: change comparison to SWAPIMODE.HVSCI once we have
+        # some HVSCI data
+        if not np.all(mode[index] == SWAPIMODE.HVENG):
+            logging.debug("MODE is not HVSCI")
+            bad_data_start_indices.append(index)
+    # TODO: add checks for checksum
+
+    # Get bad data sweep start indices and create
+    # sweep indices.
+    # Eg.
+    # From this: [0 1]
+    # To this: [[ 0  1  2  3  4  5  6  7  8  9 10 11]
+    # [12 13 14 15 16 17 18 19 20 21 22 23]]
+    cycle_start_indices = np.unique(bad_data_start_indices)
+    bad_cycle_indices = np.array(
+        [np.arange(n * 12, (n + 1) * 12) for n in cycle_start_indices]
+    ).reshape(-1)
+    return bad_cycle_indices
+
+
+def decompress_count(count_data: np.ndarray, compression_flag: np.ndarray):
+    """Decompress counts based on compression indicators.
+
+    Decompression algorithm:
+    There are 3 compression regions:
+
+    |    1) 0 <= value <=65535
+    |    2) 65536 <= value <= 1,048,575
+    |    3) 1,048,576 <= value
+
+    Pseudocode:
+
+    | if XXX_RNG_ST0 == 0:          # Not compressed
+    |    actual_value = XXX_CNT0
+    | elif (XXX_RNG_ST0==1 && XXX_CNT0==0xFFFF):    # Overflow
+    |    actual_value = <some constant that indicates overflow>
+    | elif (XXX_RNG_ST0==1 && XXX_CNT0!=0xFFFF):
+    |    actual_value = XXX_CNT0 * 16
+
+    Parameters
+    ----------
+    count_data : numpy.ndarray
+        Array with counts.
+    compression_flag : numpy.ndarray
+        Array with compression indicators.
+
+    Returns
+    -------
+    numpy.ndarray
+        Array with decompressed counts.
+    """
+    # Decompress counts based on compression indicators
+    # If 0, value is already decompressed. If 1, value is compressed.
+    # If 1 and count is 0xFFFF, value is overflow.
+    new_count = copy.deepcopy(count_data)
+    # This below line gives back row and column index for
+    # compressed flags. Eg.
+    # (array([0, 1, 2, 2]), array([ 4,  1, 20, 51]))
+    compressed_count_indices = np.where(compression_flag == 1)
+
+    for row, col in zip(*compressed_count_indices):
+        if count_data[row, col] == 0xFFFF:  # Overflow
+            new_count[row, col] = -1
+        elif count_data[row, col] != 0xFFFF:
+            new_count[row, col] = count_data[row, col] * 16
```

Comment:
```suggestion
    compressed_indices = compression_flag == 1
    new_count[compressed_indices] *= 16
    # if the data was compressed and the count was 0xFFFF, mark it as an overflow
    new_count[compressed_indices & (count_data == 0xFFFF)] = -1
```
For the -1 case, you might want to make sure the dtype of the input array and make sure it isn't an unsigned array.